libkbfs/md_ops.go

```diff
@@ -383,14 +404,22 @@ func (md *MDOpsStandard) processRange(ctx context.Context, id TlfID,
 			}
 		}
 
-		irmd, err := md.processMetadataWithID(ctx, id, bid, handle, r)
+		irmd, c, err := md.processMetadataWithID(ctx, id, bid, handle, r)
```

Comment:
I think a better, more go-like way would be to leave `processMetadata` the way it is, and then do the parallelization directly in `processRange`, copying the approach in `fbo.doBlockPuts`:

```
rmdsChan := make(chan *RootMetadataSigned, len(rmdses))
irmdChan := make(chan ImmutableRootMetadata, len(rmdses))
...
wg.Add(numWorkers)
worker := func() {
  defer wg.Done()
  for rmds := range rmdsesChan {
      .... // do everything but CheckValidSuccessor
      irmd, err := md.processMetadataWithID(..., rmds)
      // handle error if necessary
      irmdChan <- irmd
  }
}
for i := 0; i < numWorkers; i++ {
  go worker()
}
// dump rmdses into rmdsChan
wg.Wait()
// Check for errors
// Copy irmdChan into an array, then sort (or even insert in the right place using the MetadataRevision)
// Loop through and do CheckValidSuccessor
```